pkg/provider/provider.go

```diff
@@ -403,13 +403,21 @@ func (k *kubeProvider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (
 		// a resource that was created out-of-band. In this case, we do not add the `managed-by` label here, as doing
 		// so would result in a persistent failure to import due to a diff that the user cannot correct.
 		if metadata.HasManagedByLabel(oldInputs) {
-			metadata.SetManagedByLabel(newInputs)
+			_, err = metadata.TrySetManagedByLabel(newInputs)
```

Comment:
Why do you need `succeeded` if you ignore it on all usages?

Reply:
It makes it obvious that it _can_ fail.